primitives/sr-primitives/src/lib.rs

```diff
@@ -451,6 +436,27 @@ impl From<&'static str> for DispatchError {
 	}
 }
 
+/// This type specifies the outcome of dispatching a call to a module.
+///
+/// In case of failure an error specific to the module is returned.
```

Comment:
I would add a note that even in case of failure the extrinsic is included in the block and all the state changes done in `pre_dispatch` and `post_dispatch` are applied. Namely for instance the sender pays fees.